pywemo/ouimeaux_device/dimmer.py

```diff
@@ -0,0 +1,39 @@
+from . import Device
+
+
+class Dimmer(Device):
+
+    def set_state(self, state):
+        """
+        Set the state of this device to on or off.
+        """
+        self.basicevent.SetBinaryState(BinaryState=int(state))
+        self._state = int(state)
+
+    def off(self):
+        """
+        Turn this device off. If already off, will return "Error".
+        """
+        return self.set_state(0)
+
+    def on(self):
+        """
+        Turn this device on. If already on, will return "Error".
+        """
+        return self.set_state(1)
+
+    def toggle(self):
+        """
+        Toggle the dimmer's state.
+        """
+        return self.set_state(not self.get_state())
+		
+    def set_brightness(self, brightness):
+        """
+        Set the brightness of this device to an integer between 1-100.
+        """
+        self.basicevent.SetBinaryState(brightness=int(brightness))
+        self._brightness = int(brightness)
```

Comment:
It's possible you'll also need to update the on off state - so that the local on/off state works.

Easy to try - set the brightness to '0'  and then call `get_state()` and check it comes back as `0`, then set the brightness to '100'  and then call `get_state()` and check it comes back as `1`.

Assuming that doesn't work - you can just do something like
`0 if brightness == 0 else 1`